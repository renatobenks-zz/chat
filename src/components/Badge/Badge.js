// @flow

import * as React from 'react';
import styled, { css } from 'styled-components';

import Circle from './Circle/Circle';

import type { StyleProp } from '../../TypeDefinition';

const Wrapper = styled.div`
  ${props =>
    (props.top || props.left || props.bottom || props.right) &&
    css`
      position: absolute;
    `};

  width: ${props => props.size}px;
  height: ${props => props.size}px;
  display: flex;
  justify-content: center;
  align-items: center;

  ${props =>
    props.top &&
    css`
      margin-top: ${({ top }) => (top !== true ? `${top}px` : '100%')};
    `};

  ${props =>
    props.left &&
    css`
      margin-left: ${({ left }) => (left !== true ? `${left}px` : '100%')};
    `};

  ${props =>
    props.bottom &&
    css`
      margin-bottom: ${({ bottom }) => (bottom !== true ? `${bottom}px` : '100%')};
    `};

  ${props =>
    props.right &&
    css`
      margin-right: ${({ right }) => (right !== true ? `${right}px` : '100%')};
    `};

  > svg {
    width: 100%;
  }
`;

type Props = {
  size?: number,
  children: React.Element<typeof Circle>,
  top?: number | boolean,
  left?: number | boolean,
  right?: number | boolean,
  bottom?: number | boolean,
  style?: StyleProp,
};

const Badge = ({ children, style, ...props }: Props) => {
  return (
    <Wrapper {...props} style={style}>
      {children}
    </Wrapper>
  );
};

Badge.Circle = Circle;

Badge.defaultProps = {
  size: 20,
};

export default Badge;
