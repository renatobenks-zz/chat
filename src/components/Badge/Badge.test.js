import React from 'react';

import { render } from '../../../test/helpers';

import Badge from './Badge';

const renderer = component => render(component, { withTheme: true });

it('badge with children', () => {
  expect(
    renderer(
      <Badge>
        <span>badge</span>
      </Badge>,
    ),
  ).toMatchSnapshot();
});

it('badge with no children', () => {
  expect(renderer(<Badge />)).toMatchSnapshot();
});

it('badge with style', () => {
  const style = {
    justifyContent: 'flex-start',
  };

  expect(renderer(<Badge style={style} />)).toMatchSnapshot();
});

it('badge with custom size', () => {
  expect(renderer(<Badge size={60} />)).toMatchSnapshot();
});

it('badge with default positions', () => {
  expect(renderer(<Badge top />)).toMatchSnapshot();
  expect(renderer(<Badge left />)).toMatchSnapshot();
  expect(renderer(<Badge bottom />)).toMatchSnapshot();
  expect(renderer(<Badge right />)).toMatchSnapshot();
});

it('badge with custom positions', () => {
  expect(renderer(<Badge top={10} />)).toMatchSnapshot();
  expect(renderer(<Badge left={10} />)).toMatchSnapshot();
  expect(renderer(<Badge bottom={10} />)).toMatchSnapshot();
  expect(renderer(<Badge right={10} />)).toMatchSnapshot();
});
