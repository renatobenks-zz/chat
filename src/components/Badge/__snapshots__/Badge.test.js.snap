// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`badge with children 1`] = `
<div
  className="sc-bwzfXH gRMGFR"
  size={20}
>
  <span>
    badge
  </span>
</div>
`;

exports[`badge with custom positions 1`] = `
<div
  className="sc-bwzfXH hmiDgM"
  size={20}
/>
`;

exports[`badge with custom positions 2`] = `
<div
  className="sc-bwzfXH eBIpuq"
  size={20}
/>
`;

exports[`badge with custom positions 3`] = `
<div
  className="sc-bwzfXH dMXhtB"
  size={20}
/>
`;

exports[`badge with custom positions 4`] = `
<div
  className="sc-bwzfXH ewTNnx"
  size={20}
/>
`;

exports[`badge with custom size 1`] = `
<div
  className="sc-bwzfXH kgbHKq"
  size={60}
/>
`;

exports[`badge with default positions 1`] = `
<div
  className="sc-bwzfXH iENjUu"
  size={20}
/>
`;

exports[`badge with default positions 2`] = `
<div
  className="sc-bwzfXH cWmCLJ"
  size={20}
/>
`;

exports[`badge with default positions 3`] = `
<div
  className="sc-bwzfXH dYfbcO"
  size={20}
/>
`;

exports[`badge with default positions 4`] = `
<div
  className="sc-bwzfXH NGCRN"
  size={20}
/>
`;

exports[`badge with no children 1`] = `
<div
  className="sc-bwzfXH gRMGFR"
  size={20}
/>
`;

exports[`badge with style 1`] = `
<div
  className="sc-bwzfXH gRMGFR"
  size={20}
  style={
    Object {
      "justifyContent": "flex-start",
    }
  }
/>
`;
